README-automountkey.md

```diff
@@ -0,0 +1,88 @@
+Automountkey module
+=====================
+
+Description
+-----------
+
+The automountkey module allows the addition and removal of keys within an automount map. 
+
+It is desgined to follow the IPA api as closely as possible while ensuring ease of use.
+
+
+Features
+--------
+* Automount key management
+
+Supported FreeIPA Versions
+--------------------------
+
+FreeIPA versions 4.4.0 and up are supported by the ipaautomountkey module.
+
+Requirements
+------------
+**Controller**
+* Ansible version: 2.8+
+
+**Node**
+* Supported FreeIPA version (see above)
+
+
+Usage
+=====
+
+Example inventory file
+
+```ini
+[ipaserver]
+ipaserver.test.local
+```
+
+
+Example playbook to ensure presence of an automount map:
+
+```yaml
+---
+- name: Playbook to add an automount map
+  hosts: ipaserver
+  become: true
+
+  tasks:
+  - name: create key TestKey
+    ipaautomountkey:
+      ipaadmin_password: SomeADMINpassword
+      location: TestLocation
+      mapname: TestMap
+      key: TestKey
+      info: 192.168.122.1:/exports
+      state: present
+
+  - name: ensure key TestKey is absent
+    ipaautomountkey:
+      ipaadmin_password: SomeADMINpassword
+      location: TestLocation
+      mapname: TestMap
+      key: TestKey
+      state: absent
+```
+
+Variables
+=========
+
+ipaautomountkey
+-------
+
+Variable | Description | Required
+-------- | ----------- | --------
+`ipaadmin_principal` | The admin principal is a string and defaults to `admin` | no
+`ipaadmin_password` | The admin password is a string and is required if there is no admin ticket available on the node | no
+`mapname` \|  `map` \| `automountmapname` \| `automountmap` | Map the key belongs to | yes 
+`location` \| `automountlocationcn` \| `automountlocation` | Location name. | yes 
+`key` \| `name` \| `automountkey` | Automount key to manage | yes 
```

Comment:
Please, use `name` as the first alias, and position this line right below `ipaadmin_password`.

Reply:
I'm not sure about this, in the cli and api its called 'key' so I think we should go with that as the first option. In fact neither of them alias it to 'name' at all so it might make more sense just to strip out 'name' entirely, except that makes it inconsistent with every other module in not having a 'name' parameter.   Thoughts?

Reply:
I suggested `name` to be first for consistency with other modules, but, in this case, it is misleading, as the key name is composed by `mapname` and `location`, and not by `key`.